packages/valory/skills/termination_abci/tests/test_behaviours.py

```diff
@@ -0,0 +1,547 @@
+                initial_data=dict(safe_contract_address=SAFE_ADDRESS),
+                ok_reqs=[
+                    _mock_get_service_owner_request,
+                    _mock_get_zero_transfer_events_request,
+                    _mock_get_removed_owner_events_request,
+                    _mock_get_owners_request,
+                    *[_mock_get_remove_owner_data_request] * (_NUM_SAFE_OWNERS - 1),
+                    _mock_get_swap_owner_data_request,
+                ],
+                err_reqs=[_mock_get_tx_data_request],
+                expected_logs=[_MULTISEND_ERR_LOG],
+            ),
+            BehaviourTestCase(
+                name="agent fails to get a hash for the multisend tx",
+                initial_data=dict(safe_contract_address=SAFE_ADDRESS),
+                ok_reqs=[
+                    _mock_get_service_owner_request,
+                    _mock_get_zero_transfer_events_request,
+                    _mock_get_removed_owner_events_request,
+                    _mock_get_owners_request,
+                    *[_mock_get_remove_owner_data_request] * (_NUM_SAFE_OWNERS - 1),
+                    _mock_get_swap_owner_data_request,
+                    _mock_get_tx_data_request,
+                ],
+                err_reqs=[_mock_get_raw_safe_transaction_hash_request],
+                expected_logs=[_SAFE_HASH_ERR_LOG],
+            ),
+            BehaviourTestCase(
+                name="agent completes the whole flow",
+                initial_data=dict(safe_contract_address=SAFE_ADDRESS),
+                ok_reqs=[
+                    _mock_get_service_owner_request,
+                    _mock_get_zero_transfer_events_request,
+                    _mock_get_removed_owner_events_request,
+                    _mock_get_owners_request,
+                    *[_mock_get_remove_owner_data_request] * (_NUM_SAFE_OWNERS - 1),
+                    _mock_get_swap_owner_data_request,
+                    _mock_get_tx_data_request,
+                    _mock_get_raw_safe_transaction_hash_request,
+                ],
+                err_reqs=[],
+                expected_logs=[_SUCCESS_LOG],
+            ),
+        ],
+    )
+    def test_run(self, test_case: BehaviourTestCase, caplog: LogCaptureFixture) -> None:
+        """Test the happy path of termination."""
+        self.fast_forward(data=test_case.initial_data)
+        self.behaviour.act_wrapper()
+
+        # apply the OK mocks first
+        for ok_req in test_case.ok_reqs:
+            ok_req(self)
+
+        # apply the failing mocks
+        for err_req in test_case.err_reqs:
+            err_req(self, error=True)
+
+        # check that the expected logs appear
+        for expected_log in test_case.expected_logs:
+            assert expected_log in caplog.text
+
+        if len(test_case.err_reqs) == 0:
+            # no mocked requests fail,
+            # the behaviour should complete
+            self.complete()
```

Comment:
is there a reason not to run test on failure with a pytest.raises? Can't you treat them just the same as the other context managers and put these on the test case?

Reply:
We don't raise an exception here, we just log that there's an error and continue.